python/lsst/pipe/tasks/colorterms.py

```diff
@@ -19,130 +19,157 @@
+    secondary = Field(dtype=str, doc="name of secondary filter")
+    c0 = Field(dtype=float, default=0.0, doc="Constant parameter")
+    c1 = Field(dtype=float, default=0.0, doc="First-order parameter")
+    c2 = Field(dtype=float, default=0.0, doc="Second-order parameter")
+
+    # the following is desired to allow positional data, but is impossible due to DM-2381
+    # def __init__(self, primary, secondary, c0=0.0, c1=0.0, c2=0.0):
+    #     """!Construct a Colorterm
+
+    #     This overrides the default constructor for improved error detection
+    #     and to allow positional arguments
+    #     """
+    #     Config.__init__(self, primary=primary, secondary=secondary, c0=c0, c1=c1, c2=c2)
+
+    def transformSource(self, source):
+        """!Transform the brightness of a source
+
+        @param[in] source  source whose brightness is to be converted; must support get(filterName)
+                    (e.g. source.get("r")) method, as do afw::table::Source and dicts.
+        @return the transformed source magnitude
         """
-        return Colorterm._activeColorterms.get(band) if Colorterm._activeColorterms else None
+        return self.transformMags(source.get(self.primary), source.get(self.secondary))
 
-    @staticmethod
-    def transformSource(band, source, reverse=False, colorterms=None):
-        """!Transform the magnitudes in *source* to the specified *band* and return it.
-
-        The *source* must support a get(band) (e.g. source.get("r")) method, as do afw::Source and dicts.
-        Use the colorterms (or the cached set if colorterms is None); if no set is available,
-        return the *band* flux.
-        If reverse is True, return the inverse transformed magnitude
-
-        @warning reverse is not yet implemented
+    def transformMags(self, primary, secondary):
+        """!Transform primary and secondary magnitudes to a magnitude
+        
+        @param[in] primary  magnitude in primary filter
+        @param[in] secondary  magnitude in secondary filter
+        @return the transformed magnitude
         """
-        if not colorterms:
-            colorterms = Colorterm._activeColorterms
+        color = primary - secondary
+        return primary + self.c0 + color*(self.c1 + color*self.c2)
 
-        if not colorterms:
-            return source.get(band)
+    def propagateFluxErrors(self, primaryFluxErr, secondaryFluxErr):
+        return np.hypot((1 + self.c1)*primaryFluxErr, self.c1*secondaryFluxErr)
 
-        ct = colorterms[band]
-        
-        return Colorterm.transformMags(band, source.get(ct.primary), source.get(ct.secondary),
-                                       reverse, colorterms)
 
-    @staticmethod
-    def transformMags(band, primary, secondary, reverse=False, colorterms=None):
-        """!Transform the magnitudes *primary* and *secondary* to the specified *band* and return it.
+class ColortermDictConfig(Config):
+    """!A config containing dict: a dict of filterName: Colorterm
```

Comment:
While it's more explicit, I actually find that mentioning the name of the config field here makes the documentation a bit harder to read, especially with two colons that work differently.  I'd just say "A config containing a mapping from filterName to Colorterm".  Same with `ColortermLibraryConfig`.